src/Game.js

```diff
@@ -0,0 +1,80 @@
+class Game {
+	constructor(frame = new Frame()) {
```

Comment:
You may want to use class dependency injection rather than object dependency injection. Something like:
```js
constructor(frameClass = Frame, finalFrameClass = FinalFrame) {
  this.frameClass = frameClass;
  this.finalFrameClass = finalFrameClass;
  // ...
}
```